File: internal/routers/ai.go
package routers

import (
	"github.com/fluent-manager/fluent-manager/internal/middleware"
	"github.com/gin-gonic/gin"
)

func registerAIRoutes(authed *gin.RouterGroup, h *allHandlers) {
	ai := authed.Group("/ai-settings")
	{
		ai.GET("", middleware.RequirePermission("ai_settings", "read"), h.AI.GetSettings)
		ai.PUT("", middleware.RequirePermission("ai_settings", "update"), h.AI.UpdateSettings)
		ai.POST("/log-sample-analysis", middleware.RequirePermission("ai_settings", "read"), h.AI.AnalyzeLogSample)
	}
}

File: web/frontend/src/api/ai.ts
import api from './client'

export const getAISettings = () => api.get('/ai-settings')
export const updateAISettings = (data: Record<string, any>) => api.put('/ai-settings', data)
export const analyzeLogSample = (data: Record<string, any>) => api.post('/ai-settings/log-sample-analysis', data)
